bin/install.sh

```diff
@@ -22,21 +26,57 @@ COLOR_RESET="\e[0m"
 BASE_DIR=$(dirname "${SCRIPT_DIRECTORY}")
 
 function print() {
-  if [[ $# = 0 ]]; then
+  if [[ $# == 0 ]]; then
     echo
-  elif [[ $# = 1 ]]; then
+  elif [[ $# == 1 ]]; then
     local message="${1}"
     echo "${message}"
-  elif [[ $# = 2 ]]; then
+  elif [[ $# == 2 ]]; then
     local color="${1}"
     local message="${2}"
     echo -e "${color}${message}${COLOR_RESET}"
   fi
 }
 
+function usage() {
```

Comment:
Usually the usage string looks like: `Usage: command --required-option [--optional-options] [-h|--help] <filename>` so in this case I would expcet: `Usage: $(basename "$0") [--all] [--scanners] [--hooks] [--demo-apps] [-h]`.

A good pattern is to store usage and help in variables, eg.:

```
USAGE="Usage: $(basename "$0") <hostname> <jenkins_secret>"
HELP=$(cat <<- EOT
Here tell the user that w/o arguments the installatiuon is interactive.

Options:
  --all                  Explanation...
  --scanners       Explanation
  --hooks            Explanation
  --demo-apps   Explanation
  -h|--help          Explanation

Examples:

Add some examples.
EOT
)
```

These variables you can use in a function to print help:

```
print_help() {
    echo "$USAGE"
    echo
    echo "$HELP"
    echo
}
```

The `USAGE` you can print before any non-zero exit as hint.